experimental/filesystem-backends.mdx

```diff
@@ -0,0 +1,332 @@
+    # for private repository access
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_USERNAME=...
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_PASSWORD=...
+    ```
+
+  </Tab>
+  <Tab title="Configuration Yaml">
+
+    ```yaml
+    experimental:
+      filesystem_storage:
+        enabled: true
+
+    storage:
+      type: git
+      git:
+        repository: "https://github.com/predictab.le/config.git"
+        ref: "main"
+        poll_interval: "30s"
+        authentication:
+          basic:
+            username: ...
+            password: ...
+          token:
+            access_token: ...
+    ```
+
+  </Tab>
+</Tabs>
+
+#### Authentication
+
+The `git` type backend supports both `basic` and `token` based authentication schemes.
+These sections are mutually exclusive. Flipt will fail to initialize with a validation error if you attempt to configure both.
+
+<Note>
+  GitHub PAT (Personal Access Tokens) should be supplied using `basic`
+  authentication as the password. Use your GitHub username in the `username`
+  field.
+</Note>
+
+## Flag State Configuration
+
+Each of Flipts filesystem backends expects you to represent your feature flag configuration via a set of yaml files.
+These files declaratively define what flags, segments, variants etc. exist and in what configuration.
+
+### Locating Flag State
+
+Flipts filesystem backends allow you to define feature flags alongside other configuration in a shared directory or repository.
+In order to identify which files are intended for Flipt, it uses a naming scheme to index which are flag state files.
+By default, Flipt will look for the following filename patterns when identifying files to attempt to parse as Flipt state:
+
+- `**/features.yaml`
+- `**/features.yml`
+- `**/*.features.yaml`
+- `**/*.features.yml`
+
+Meaning, any file named `features.yaml`, `features.yml` or with either extension `.features.yaml` or `.features.yml` is considered.
+This is recursively from the root of your target directory or repository.
+
+If this naming convention does not work for you, for whatever reason, it can be overriden by creating a file named `.flipt.yml` in the root of your target directory tree.
+This file will be used to instruct Flipt on how to index your directory tree and find flag state files:
+
+```yaml
+version: "1.0"
+include:
+  - "**/features.yaml"
+  - "**/features.yml"
+  - "**/*.features.yaml"
+  - "**/*.features.yml"
+exclude: []
+```
+
+The index file contains two lists `include` and `exclude`. These can contain specific paths or Glob matching patterns.
+The indexing process first matches the `include` section and then filters that set by the `exclude` section.
+
+### Defining Flag State
+
+<Info>
+  Flipt flag state file format has been taken directly from Flipts existing
+  [import and export](/configuration/storage#import-export) flag state format.
+  You can run `flipt export` and add / commit the result to a directory or git
+  repository to get started quickly.
```

Comment:
Maybe here we can mention if they already have data in a db then they can do this? Something indicating about migrating data from flipt backed by a db already essentially.

Reply:
👍 I will make it clearer that this can be performed against an existing instance with data in it.